Catappult/appcoins-wallet-android
# [feature/APPC-2709] Influencer promo code

**What does this PR do?**

implements the option to add a promo code and receive bonus from that promo code in the respective transactions

**Database changed?**

   Yes 

**Where should the reviewer start?**

- PromoCodeBottomSheetFragment.kt
- PromoCodeRepository.kt
- SettingsFragment.kt

**How should this be manually tested?**

- add an invalid promo code
- add an active promo code -> "megapack"
- add an expired promo code ->"megapack2"
- check the bonus in the payment popup, with and without an active promo code
- make a payment, with and without the promo code
- check the transactions list 

**What are the relevant tickets?**

  https://aptoide.atlassian.net/browse/APPC-2709


**Code Review Checklist**
